### PR TITLE
wmr - Refactoring API routes

### PR DESCRIPTION
**Pull request by jtasonye**
**Changes made in PR:**
- Mirrored the api route name changes on the UI to follow RESTful conventions

### CODE REVIEW

File: src/services/residentLife/ResidentStaff.ts
Comment (EjPlatzer) [inline src/services/residentLife/ResidentStaff.ts:33]: String interpolation is not the best tool for adding search params to a URL, because there's a lot of characters that have to be specially encoded. Browsers provide the `URLSearchParams` class to make working with search params easier, but in 360 we've made a simple utility function to handle most cases:
```suggestion
  http.get(`Housing/ras${http.toQueryString({hallId, roomNumber})`);
```

**Review by EjPlatzer — CHANGES_REQUESTED** [review]
One small note on handling search params in the query string of a URL.

File: src/services/residentLife/ResidentStaff.ts
Comment (RossClark01) [reply src/services/residentLife/ResidentStaff.ts]: Updated accordingly

**Review by RossClark01 — APPROVED** [review]
Thanks for starting this sir! Adjusted routes a bit more to account for a few more changes. I have a few more points to address on the API side so we can merge this into WMRA after changes are squared away

**Comment by jtasonye** [comment]
Thanks for catching that Ross!

**Comment by RossClark01** [comment]
@jtasonye added a few commits here, can you take a look and make sure everything looks good. Once done we should be able to merge this into our WMRA branch and get it into SP. The API side has another PR open that once reviewed can be merged into WMRA then SP for further user testing

**Comment by jtasonye** [comment]
@RossClark01 LGTM!

**Review by russtuck — COMMENTED** [review]
Why do we have "post" instead of "put" here?  

It looks like all the post commands are idempotent and modify existing data.  That means that a put would be more appropriate.  Post should generally only be used when asking the API server to create something new and assign it a unique URL.

**Pull request merged**
